File: 30 Days of Code/Day 8 Dictionaries and Maps/Solution.cpp
// https://www.hackerrank.com/challenges/30-dictionaries-and-maps/problem

#include <cmath>
#include <cstdio>
#include <vector>
#include <iostream>
#include <algorithm>
#include <map>
using namespace std;


int main() {
    /* Enter your code here. Read input from STDIN. Print output to STDOUT */   
    map<string, string> phoneBook;
    int noofEntries;
    cin>>noofEntries;
    for(int entry=0; entry < noofEntries; ++entry){
        string name, phoneNumber;
        cin>>name>>phoneNumber;
        phoneBook.insert(pair<string, string>(name, phoneNumber));
    }
    while(!cin.eof()){
        string name;
        cin>>name;
        map<string,string>::iterator it;
        it = phoneBook.find(name);
        if(it != phoneBook.end()){
            cout<<name<<"="<<phoneBook[name]<<endl;
        }
        else{
            cout<<"Not found"<<endl;
        }
    }
    return 0;
}

File: 30 Days of Code/Day 8 Dictionaries and Maps/Solution.py
# https://www.hackerrank.com/challenges/30-dictionaries-and-maps/problem

# Enter your code here. Read input from STDIN. Print output to STDOUT

if __name__ == "__main__":
    phoneBook = {}
    NoofEntries = int(input())
    for entry in range(NoofEntries):
        content = input()
        name, phoneNumber = content.split()
        phoneBook[name] = phoneNumber

    while True:
        try:
            name = input()  
            if name in phoneBook:
                print(name,phoneBook[name],sep="=")
            else:
                print("Not found")     
        except EOFError:
            break